recipes/cuda-nvcc/meta.yaml

```diff
@@ -0,0 +1,65 @@
+{% set name = "cuda-nvcc" %}
+{% set version = "12.0.76" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/cuda_nvcc/{{ platform }}/cuda_nvcc-{{ platform }}-{{ version }}-archive.{{ extension }}
+
+  sha256: 160ca9f8828f53daa4b2066c5361aacb45fa2575885f70c223cda1d11df53d6f  # [linux64]
+  sha256: dc5bd7eac57b32a9645c0c23c6a1ba8a9533e846eeec31414e1aa4230afee2f8  # [ppc64le]
+  sha256: 85c8c643ea2cfa398e1b5ffcd597f5bb3b738526c9e3fb8e39dd909c55345f36  # [aarch64]
+  sha256: f67406f44263f0e4a110a8cf0200bff1d3d0e32917db8a7c9e417a87eb7ca372  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+  run_exports:
+    strong:
+      - __glibc >=2.17        # [linux]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  host:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+  run:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+  run_constrained:
+    - gcc_impl_{{ target_platform }} >=6,<13        # [linux]
+
+test:
+  requires:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - cuda-cudart-dev
+  files:
+    - test.cu
+  commands:
+    - nvcc --version
```

Comment:
we should add `nvcc --ccbin "${CXX} -arch=native test.cu` to validate that the `__nvcc_device_query` executable that nvcc calls is also packaged

Reply:
How should we handle cross-compiling use cases?

Reply:
I think what I am really asking for is a check that the expected internal executables exist in the package ( `fatbinary`, `nvlink`, `cudafe++`, `__nvcc_device_query`, etc ). 

That would avoid any execution of `nvcc`

Reply:
Will this still work even when there is no GPU?

Reply:
> Will this still work even when there is no GPU?

Yes. The compiler will generate a warning ( `nvcc warning : Cannot find valid GPU for '-arch=native', default arch is used` ) and continue to execute.